### PR TITLE
Fix DLQ fails to start due to read 1 byte file

### PR DESCRIPTION
**Pull request by kaisecheng**
## Release notes

Fix DLQ fails to start due to read 1 byte file

## What does this PR do?

This commit ignores DLQ files that contain only the version number. These files have no content and should be skipped. 

## Why is it important/What is the impact to the user?

Mapping 1 byte DLQ files to buffer causes java.lang.IllegalArgumentException: newPosition < 0: (-1 < 0)
User is unable to start the pipeline using dead_letter_queue input

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files (and/or docker env variables)
- [ ] I have added tests that prove my fix is effective or that my feature works

## Author's Checklist

- [x] manually test the 1 byte file and can start Logstash 

## How to test this PR locally

follow the reproducer of #14599

## Related issues

- Fixed: #14599

### CODE REVIEW

File: logstash-core/src/test/java/org/logstash/common/io/DeadLetterQueueReaderTest.java [hunk 1]
```diff
@@ -543,6 +543,23 @@ public void testSeekByTimestampWhenAllSegmentsAreDeleted() throws IOException, I
         }
     }
 
+    @Test
+    public void testSeekByTimestampWhenSegmentIs1Byte() throws IOException, InterruptedException {
+        final long startTime = System.currentTimeMillis();
+        com.google.common.io.Files.touch(dir.resolve("1.log").toFile());
```
Comment (andsel) [inline logstash-core/src/test/java/org/logstash/common/io/DeadLetterQueueReaderTest.java:549]: I think it should insert one byte in the offending segment, the [version byte](https://github.com/elastic/logstash/blob/46babd6041eba476ba61abd5c3ce5486dc7f2997/logstash-core/src/main/java/org/logstash/common/io/RecordIOWriter.java#L99), to be adherent to reality

File: logstash-core/src/main/java/org/logstash/common/io/DeadLetterQueueUtils.java [hunk 2]
```diff
@@ -41,10 +41,7 @@ static int extractSegmentId(Path p) {
     }
 
     static Stream<Path> listFiles(Path path, String suffix) throws IOException {
-        try(final Stream<Path> files = Files.list(path)) {
-            return files.filter(p -> p.toString().endsWith(suffix))
-                    .collect(Collectors.toList()).stream();
-        }
+        return Files.list(path).filter(p -> p.toString().endsWith(suffix));
```
Comment (andsel) [inline logstash-core/src/main/java/org/logstash/common/io/DeadLetterQueueUtils.java:44]: It's not clear to me why the removal of try-finally on a closable resource (the stream of file names) is functional to this fix.
At the end it creates a new stream of paths from  a stream of paths, however the originating stream resource should be always be cleanly closed.

**Review by andsel — CHANGES_REQUESTED** [review]
The fix is effective, I've tested on `main` branch, left a couple of doubts

File: logstash-core/src/main/java/org/logstash/common/io/DeadLetterQueueUtils.java (hunk 2 above)
Comment (kaisecheng) [reply logstash-core/src/main/java/org/logstash/common/io/DeadLetterQueueUtils.java:44]: You are right. It is a file IO and should be try-finally, will revert it

**Review by andsel — APPROVED** [review]
LGTM

**Comment by kaisecheng** [comment]
@logstashmachine backport 8.4

**Comment by kaisecheng** [comment]
@logstashmachine backport 8.5

**Pull request merged**